frontends/ast/simplify.cc

```diff
@@ -2218,6 +2218,27 @@ bool AstNode::simplify(bool const_fold, bool at_zero, bool in_lvalue, int stage,
 			children.push_back(node);
 			did_something = true;
 		}
+		else if (str == "buf" || str == "not")
+		{
+			AstNode *input = children_list.back();
+			if (str == "not")
+				input = new AstNode(AST_BIT_NOT, input);
+
+			std::vector<AstNode *> assigns;
```

Comment:
Arguably the `newNode` should be an `AST_GENBLOCK` containing all of the new assignments, rather than adding one assignment in place and the rest at the end of the module.

Reply:
Ah, thats's what I was looking for! I tried `AST_BLOCK`, but that failed.